variant-remapping-automation/prepare_remapping_table.py

```diff
@@ -0,0 +1,351 @@
+    with get_metadata_connection_handle('production_processing', private_config_xml_file) as pg_conn:
+        query = f"select * from eva_progress_tracker.remapping_tracker where release_version={release_version}"
+        return get_all_results_for_query(pg_conn, query)
+
+
+"""
+For each assembly, get the total number of ssids to remap, by reading the accessioning report for all the studies 
+which needs to be remapped
+"""
+
+
+def get_asm_no_ss_ids(asm_studies):
+    asm_ssids = {}
+    for asm, studies in asm_studies.items():
+        no_of_ss_ids_in_asm = 0
+        for study in studies:
+            # get path to accessioned file on both noah and codon
+            noah_file_path, codon_file_path = get_accession_file_path_for_study(study)
+            if noah_file_path:
+                no_of_ss_ids_in_asm = no_of_ss_ids_in_asm + get_no_of_ss_ids_in_file(noah_file_path)
+            if codon_file_path:
+                no_of_ss_ids_in_asm = no_of_ss_ids_in_asm + get_no_of_ss_ids_in_file(codon_file_path)
+
+        logger.info(f"No of ss ids in assembly {asm} are {no_of_ss_ids_in_asm}")
+        asm_ssids[asm] = no_of_ss_ids_in_asm
+
+    return asm_ssids
+
+
+"""
+Read a file to figure out the number of ss ids in that file
+skip line start with #
+read line where 3rd entry starts with ss (assuming all files have same sequence of headers and ss id is always 3rd)
+"""
+
+
+def get_no_of_ss_ids_in_file(path):
+    no_of_ss_ids_in_file = 0
+    with gzip.open(path, 'rt') as f:
+        for line in f:
+            if line.startswith('#'):
+                continue
+            elif line.split("\t")[2].startswith("ss"):
+                no_of_ss_ids_in_file = no_of_ss_ids_in_file + 1
+
+    print(f"No of ss ids in file {path} : {no_of_ss_ids_in_file}")
+    return no_of_ss_ids_in_file
+
+
+"""
+Given a study, find the accessioning report path for that study on both noah and codon
+look for a file ending with accessioned.vcf.gz (assuming only one file with the given pattern will be present)
+if more than one files are present, take the first one
+"""
+
+
+def get_accession_file_path_for_study(study):
+    noah_file = glob.glob(f"/nfs/ebi/production3/eva/data/{study}/60_eva_public/*accessioned.vcf.gz")
+    codon_file = glob.glob(f"/nfs/production/keane/eva/data/{study}/60_eva_public/*accessioned.vcf.gz")
+
+    if not noah_file and not codon_file:
+        logger.error(f"Could not find any file in Noah or Codon for Study {study}")
+    if not noah_file:
+        logger.warning(f"No file found in Noah for Study {study}")
+        noah_file = None
+    if not codon_file:
+        logger.warning(f"No file found in Codon for Study {study}")
+        codon_file = None
+
+    if noah_file and len(noah_file) > 1:
+        logger.warning(f"More than one accessioned files found in Noah for Study {study}: {noah_file}. "
+                       f"Reading {noah_file[0]}")
+        noah_file = noah_file[0]
```

Comment:
I don't think it will be that unusual to have more than one accessioned file per study, I believe we do one per analysis.  (And actually there will be occasional cases where there are different assemblies among the analyses, though I don't think we have to worry about that here...)

Reply:
updated to read from all files